### PR TITLE
Generalise ENT functions

### PR DESCRIPTION
**Pull request by tarneaux**
From [roadmap](https://github.com/bain3/pronotepy/projects/1#card-63536653)
I haven't tested ANY of these functions I changed.
It should be tested before merging.
If you don't test it before, I don't take any responsibility for any issues.

### CODE REVIEW

**Comment by JulesGrd** [comment]
I will be able to test the Orleans-Tours Ent from September 2

File: pronotepy/ent.py [hunk 1]
```diff
@@ -59,41 +59,48 @@ def atrium_sud(username, password):
     return requests.utils.cookiejar_from_dict(requests.utils.dict_from_cookiejar(session.cookies))
 
 
-def ac_reims(username, password):
+def ac_reims_and_occitanie_montpellier(username, password, orig_url, ent_login, ent_verif, pronote_verif, SERVERID, preselection):
```
Comment (bain3) [inline pronotepy/ent.py:62]: Should probably be private (add `_`)

File: pronotepy/ent.py [hunk 2]
```diff
@@ -226,24 +228,26 @@ def ac_reunion(username, password):
 
     return requests.utils.cookiejar_from_dict(requests.utils.dict_from_cookiejar(session.cookies))
 
-def ile_de_france(username, password):
+
+
+def ile_de_france_and_paris_classe_numerique(username, password, ent_login):
```
Comment (bain3) [inline pronotepy/ent.py:233]: Should be private too

**Comment by bain3** [comment]
I'm not sure how to handle this. Not being able to test these changes practically in any way really sucks. Unfortunately the ENT system is really badly designed. Good old OAuth2...

I'm not gonna merge this request for a few days until all ENTs are definitely working so we can at least attempt to somewhat verify that the code works.

**Comment by tarneaux** [comment]
After 3 hours of train I notice this is actually a bit unreadable; a ENT login rework is needed if we want to make the whole thing more readable. Right now you can't really understand what the thing does, especially for ac-reims and occitanie.
Closing this as it just makes the code worse and could cause bugs.
As you say @bain3 we need to wait until it's testable.

**Pull request closed without merging**